panic: runtime type (Iface) not found, install from pre-built package or set LLGO_ROOT

use the https://github.com/goplus/llgo/releases/download/v0.11.5/llgo0.11.5.darwin-arm64.tar.gz

```bash
Code/temp/ttt on  main [✘!+?⇕] via 🐹 v1.24.6 via 💎 v2.6.10 
❯ ls /Users/zhangzhiyang/Downloads/llgo0.11.5.darwin-arm64                                                     
bin       LICENSE   README.md runtime
(base) 
Code/temp/ttt on  main [✘!+?⇕] via 🐹 v1.24.6 via 💎 v2.6.10 
❯ export LLGO_ROOT=/Users/zhangzhiyang/Downloads/llgo0.11.5.darwin-arm64                                                     
(base) 
Code/temp/ttt on  main [✘!+?⇕] via 🐹 v1.24.6 via 💎 v2.6.10 
❯ /Users/zhangzhiyang/Downloads/llgo0.11.5.darwin-arm64/bin/llgo run main.go 
WARNING: LLGO_ROOT is not a valid LLGO root: /Users/zhangzhiyang/Downloads/llgo0.11.5.darwin-arm64
panic: runtime type (Iface) not found, install from pre-built package or set LLGO_ROOT

goroutine 1 [running]:
github.com/goplus/llgo/ssa.(*aProgram).rtNamed(0x14000221208, {0x1048b51ff, 0x5})
        /go/src/llgo/ssa/package.go:330 +0x1ac
github.com/goplus/llgo/ssa.(*aProgram).rtType(0x14000221208, {0x1048b51ff?, 0x140047c2b18?})
        /go/src/llgo/ssa/package.go:334 +0x24
github.com/goplus/llgo/ssa.(*aProgram).rtIface(...)
        /go/src/llgo/ssa/package.go:346
github.com/goplus/llgo/ssa.(*aProgram).toType(0x14000221208, {0x104abb390, 0x1400016ea50})
        /go/src/llgo/ssa/type.go:373 +0x3f8
github.com/goplus/llgo/ssa.(*aProgram).rawType(0x14000221208, {0x104abb390, 0x1400016ea50})
        /go/src/llgo/ssa/type.go:252 +0x68
github.com/goplus/llgo/ssa.(*aProgram).toNamed(0x14000221208, 0x14000166070)
        /go/src/llgo/ssa/type.go:516 +0x200
github.com/goplus/llgo/ssa.(*aProgram).toType(0x14000221208, {0x104abb3b8, 0x14000166070})
        /go/src/llgo/ssa/type.go:382 +0xd10
github.com/goplus/llgo/ssa.(*aProgram).rawType(0x14000221208, {0x104abb3b8, 0x14000166070})
        /go/src/llgo/ssa/type.go:252 +0x68
github.com/goplus/llgo/ssa.(*aProgram).toLLVMTypes(0x14000221208, 0x140019c9020, 0x2)
        /go/src/llgo/ssa/type.go:452 +0x84
github.com/goplus/llgo/ssa.(*aProgram).toLLVMFunc(0x14000221208, 0x140019cd9c0)
        /go/src/llgo/ssa/type.go:465 +0xdc
github.com/goplus/llgo/ssa.(*aProgram).FuncDecl(0x14000221208, 0x10472b858?, 0x1400701cc70?)
        /go/src/llgo/ssa/type_cvt.go:66 +0xa8
github.com/goplus/llgo/ssa.(*aPackage).NewFuncEx(0x140021a5380, {0x1400701cc70, 0x9}, 0x140019cd9c0, 0x1, 0x0, 0x0)
        /go/src/llgo/ssa/decl.go:206 +0x6c
github.com/goplus/llgo/cl.(*context).compileFuncDecl(0x14001aa7d40, 0x140021a5380, 0x14003680e00)
        /go/src/llgo/cl/compile.go:255 +0x3f0
github.com/goplus/llgo/cl.processPkg(0x14001aa7d40, 0x140021a5380, 0x140035bcf00)
        /go/src/llgo/cl/compile.go:1086 +0x2a4
github.com/goplus/llgo/cl.NewPackageEx(0x14000221208, 0x1400204a600, 0x140035bcf00, {0x140012bd950, 0x3, 0x3})
        /go/src/llgo/cl/compile.go:1033 +0x620
github.com/goplus/llgo/internal/build.buildPkg(0x140017f6680, 0x14003693c20, 0x0)
        /go/src/llgo/internal/build/build.go:792 +0x248
github.com/goplus/llgo/internal/build.buildAllPkgs(0x140017f6680, {0x1400132bf10?, 0x1048b4ec1?, 0x1049ae0b8?}, 0x0)
        /go/src/llgo/internal/build/build.go:413 +0x2e0
github.com/goplus/llgo/internal/build.Do({0x1400003c230, 0x1, 0x1}, 0x1400007e240)
        /go/src/llgo/internal/build/build.go:253 +0xcac
github.com/goplus/llgo/cmd/internal/run.runCmdEx(0x14000080ae8?, {0x1400003c230?, 0x14000080ae8?, 0x1045cc6bc?}, 0x2)
        /go/src/llgo/cmd/internal/run/run.go:79 +0x130
github.com/goplus/llgo/cmd/internal/run.runCmd(0x22?, {0x1400003c230?, 0x14000080ae8?, 0x1045cc344?})
        /go/src/llgo/cmd/internal/run/run.go:57 +0x24
main.(*Cmd_run).Main.func1({0x1400003c230?, 0x1045cc334?, 0x14000214500?})
        cmd/llgo/run_cmd.gox:27 +0x40
main.(*Cmd_run).Main.(*Command).Run__1.func2(0x14000214500?, {0x1400003c230?, 0x4?, 0x1048b4c42?})
        /root/go/pkg/mod/github.com/goplus/cobra@v1.9.12/xcmd/classfile.go:87 +0x34
github.com/goplus/cobra.(*Command).execute(0x1400021d208, {0x1400003c230, 0x1, 0x1})
        /root/go/pkg/mod/github.com/goplus/cobra@v1.9.12/command.go:1019 +0x810
github.com/goplus/cobra.(*Command).ExecuteC(0x1400021c008)
        /root/go/pkg/mod/github.com/goplus/cobra@v1.9.12/command.go:1148 +0x350
github.com/goplus/cobra.(*Command).Execute(...)
        /root/go/pkg/mod/github.com/goplus/cobra@v1.9.12/command.go:1071
github.com/goplus/cobra/xcmd.Gopt_App_Main({0x104ab85d8, 0x1400021c008}, {0x14000080e88, 0x8, 0x8})
        /root/go/pkg/mod/github.com/goplus/cobra@v1.9.12/xcmd/classfile.go:133 +0x25c
main.(*App).Main(0x1400021c008)
        cmd/llgo/main_app.gox:12 +0x2b0
main.main()
        cmd/llgo/version_cmd.gox:34 +0x28
(base) 
```

--- a/internal/env/env.go
+++ b/internal/env/env.go
@@ -97,12 +97,12 @@ func isLLGoRoot(root string) (string, bool) {
 		return "", false
 	}
 	// Check for go.mod
-	data, err := os.ReadFile(filepath.Join(root, "go.mod"))
+	data, err := os.ReadFile(filepath.Join(root, LLGoRuntimePkgName, "go.mod"))
 	if err != nil {
 		return "", false
 	}
 	// Check module name
-	if !strings.Contains(string(data), "module "+LLGoCompilerPkg+"\n") {
+	if !strings.Contains(string(data), "module "+LLGoRuntimePkg+"\n") {
 		return "", false
 	}
 	return root, true

--- a/internal/env/env_test.go
+++ b/internal/env/env_test.go
@@ -41,8 +41,10 @@ func TestLLGoRuntimeDir(t *testing.T) {
 		defer os.Setenv("LLGO_ROOT", origLLGoRoot)
 
 		tmpDir := t.TempDir()
-		goModContent := []byte("module github.com/goplus/llgo\n")
-		if err := os.WriteFile(filepath.Join(tmpDir, "go.mod"), goModContent, 0644); err != nil {
+		runtimeDir := filepath.Join(tmpDir, "runtime")
+		os.MkdirAll(runtimeDir, 0755)
+		goModContent := []byte("module github.com/goplus/llgo/runtime\n")
+		if err := os.WriteFile(filepath.Join(runtimeDir, "go.mod"), goModContent, 0644); err != nil {
 			t.Fatal(err)
 		}
 
@@ -92,8 +94,10 @@ func TestLLGoROOT(t *testing.T) {
 		defer os.Setenv("LLGO_ROOT", origLLGoRoot)
 
 		tmpDir := t.TempDir()
-		goModContent := []byte("module github.com/goplus/llgo\n")
-		if err := os.WriteFile(filepath.Join(tmpDir, "go.mod"), goModContent, 0644); err != nil {
+		runtimeDir := filepath.Join(tmpDir, "runtime")
+		os.MkdirAll(runtimeDir, 0755)
+		goModContent := []byte("module github.com/goplus/llgo/runtime\n")
+		if err := os.WriteFile(filepath.Join(runtimeDir, "go.mod"), goModContent, 0644); err != nil {
 			t.Fatal(err)
 		}
 
@@ -170,8 +174,10 @@ func TestIsLLGoRoot(t *testing.T) {
 	// Test with valid path and valid go.mod
 	t.Run("valid path and go.mod", func(t *testing.T) {
 		tmpDir := t.TempDir()
-		goModContent := []byte("module github.com/goplus/llgo\n")
-		if err := os.WriteFile(filepath.Join(tmpDir, "go.mod"), goModContent, 0644); err != nil {
+		runtimeDir := filepath.Join(tmpDir, "runtime")
+		os.MkdirAll(runtimeDir, 0755)
+		goModContent := []byte("module github.com/goplus/llgo/runtime\n")
+		if err := os.WriteFile(filepath.Join(runtimeDir, "go.mod"), goModContent, 0644); err != nil {
 			t.Fatal(err)
 		}
 

